lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.cpp

```diff
@@ -728,10 +651,77 @@ void ProxyShape::onPrimResync(SdfPath primPath, const SdfPathVector& previousPri
   constructGLImagingEngine();
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+void ProxyShape::serializeThis()
+{
+  UsdStageRefPtr stage = getUsdStage();
+
+  if(stage)
+  {
+    // Serialize our own session layer - LayerManager will handle all other layers
+    std::string serializeSessionLayerStr;
+    stage->GetSessionLayer()->ExportToString(&serializeSessionLayerStr);
```

Comment:
Hmm, very good catch! Functionally, I don't this this will create any problems right now - on file re-open, it will simply read the serialized version stored on the ProxyShape attribute, and ignore the one stored in the LayerManager - but it's obviously wasteful, and should be avoided.

I actually originally stored all layers - including anonymous session layers - on the layer manager, but went back to just storing them on the ProxyShape because it made the code much simpler... I was either having to keep attribute links (which got messy because then layer ordering suddenly mattered, and needed to be remembered), or had to do name-translation/swapping (because you can't create an anonymous layer with a specific name... you can specify a "base" name, but then the address always gets appended in front, so effectively, you can't).  However, neither of these is insurmountable, so I may go back to that model, to avoid this duplication.  It's a "purer" approach, too, in the sense that all layers are stored in the same place.

Probably wont' be able to get that by EOD, though, so... for now, the current version should at least be functional.

Reply:
Yeah I didn't think this would cause a problem, just a small wasteful storing of a string.

Will you be looking at doing the change you mentioned, save all layers into the LayerManager, before merging?